src/main/java/org/springframework/data/repository/config/RepositoryConfigurationExtension.java

```diff
@@ -65,6 +79,17 @@ <T extends RepositoryConfigurationSource> Collection<RepositoryConfiguration<T>>
 	 */
 	String getRepositoryFactoryBeanClassName();
 
+	/**
+	 * Return the annotations to scan domain types for when evaluating repository interfaces for store assignment. Modules
+	 * should return the annotations that identify a domain type as managed by the store explicitly.
+	 *
+	 * @return
+	 * @since 3.0
+	 */
+	default Collection<Class<? extends Annotation>> getIdentifyingAnnotations() {
```

Comment:
It feels a bit weird handing out identifying annotations to the outside of the config extension as the managed type scanning vs. repository scanning are located at two different abstractions.

Repository type scanning happens inside the `RepositoryConfigurationExtension`. We still need to merge repository-managed types with the ones that we discover through the classpath scan.

How about returning a `RepositoryConfigurations` object from `getRepositoryConfigurations` that wraps `Collection<RepositoryConfiguration<T>>` and provides a `ManagedTypes` object?

Doing so would allow keeping all scanning-related operations within the extension.